rcldotnet/Clock.cs

```diff
@@ -0,0 +1,261 @@
+
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_enable_ros_time_override), out native_rcl_enable_ros_time_override);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_disable_ros_time_override), out native_rcl_disable_ros_time_override);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_clock_get_now), out native_rcl_clock_get_now);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_set_ros_time_override), out native_rcl_set_ros_time_override);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_clock_add_jump_callback), out native_rcl_clock_add_jump_callback);
+            _dllLoadUtils.RegisterNativeFunction(nativeLibrary, nameof(native_rcl_clock_remove_jump_callback), out native_rcl_clock_remove_jump_callback);
+        }
+    }
+
+    public sealed class Clock
+    {
+        private readonly Dictionary<JumpCallback, JumpCallbackInternal> _registeredJumpCallbacks = new Dictionary<JumpCallback, JumpCallbackInternal>();
+
+        internal Clock(SafeClockHandle handle)
+        {
+            Handle = handle;
+        }
+
+        internal SafeClockHandle Handle { get; }
+
+        public Time Now()
+        {
+            RCLRet ret = ClockDelegates.native_rcl_clock_get_now(Handle, out TimePoint timePoint);
+
+            RCLExceptionHelper.CheckReturnValue(ret, $"{nameof(ClockDelegates.native_rcl_clock_get_now)}() failed.");
+
+            return timePoint.ToMsg();
+        }
+
+        internal RCLRet EnableRosTimeOverride()
+        {
+            RCLRet ret = ClockDelegates.native_rcl_enable_ros_time_override(Handle);
+
+            RCLExceptionHelper.CheckReturnValue(ret, $"{nameof(ClockDelegates.native_rcl_enable_ros_time_override)}() failed.");
+
+            return ret;
+        }
+
+        internal RCLRet DisableRosTimeOverride()
+        {
+            RCLRet ret = ClockDelegates.native_rcl_disable_ros_time_override(Handle);
+
+            RCLExceptionHelper.CheckReturnValue(ret, $"{nameof(ClockDelegates.native_rcl_disable_ros_time_override)}() failed.");
+
+            return ret;
+        }
+
+        internal RCLRet SetRosTimeOverride(long timePointValue)
+        {
+            RCLRet ret = ClockDelegates.native_rcl_set_ros_time_override(Handle, timePointValue);
+
+            RCLExceptionHelper.CheckReturnValue(ret, $"{nameof(ClockDelegates.native_rcl_set_ros_time_override)}() failed.");
+
+            return ret;
+        }
+
+        public void AddJumpCallback(JumpThreshold threshold, JumpCallback callback)
+        {
+            if (_registeredJumpCallbacks.ContainsKey(callback))
+            {
+                throw new CallbackAlreadyRegisteredException("Provided jump callback was already registered!");
+            }
+
+            JumpCallbackInternal callbackInternal = (timeJumpPtr, beforeJump) =>
```

Comment:
It's getting late. Note for myself or for someone to answer:

Does passing the delegate to the native side work without the GC moving the delegate object and causing calls later on to fail?